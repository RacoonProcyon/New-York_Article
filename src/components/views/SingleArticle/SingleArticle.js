import React, { Component } from 'react'
import Header from '../../Sections/Header.js'
import { connect } from "react-redux";
import { searchArticle,clearArticleData } from '../../../redux/action'
import {Link } from 'react-router-dom'

class SingleArticle extends Component {
    constructor() {
        super()
        this.state = {
            headLine: "HeadLine",
            Date: "DATE",
            leadParagraph: "Details",
            webUrl : "#"
        }
    }

    componentDidMount = () => {
        this.props.searchArticle(this.props.match.params.id)
    }

    componentDidUpdate = (prevProps, prevState) => {
        if (prevProps.singleArticle.length !== this.props.singleArticle.length) {
            this.setState({
                ...this.state,
                headLine: this.props.singleArticle[0].headline.main,
                Date: this.props.singleArticle[0].pub_date.slice(0, 10),
                leadParagraph: this.props.singleArticle[0].lead_paragraph,
                webUrl : this.props.singleArticle[0].web_url
            })
        }
    }

    componentWillUnmount=()=>{
        this.props.clearArticleData()
    }

    render() {
        return (
            <div className='d-flex flex-column align-items-center'>
                <Header />
                <div className="mt-5 col-12 col-md-8" >
                    {this.props.query?<Link to={"/"+this.props.query}>Go to result page</Link>:null}
                    <h3 className='mt-1' style={{ fontWeight: '700' }}>{this.state.headLine}</h3>
                    <span className='mt-2'>{this.state.Date}</span>
                    <p className='mt-3'>{this.state.leadParagraph}</p>
                    <a target="_blank" href={this.state.webUrl} >Read the full article</a>
                </div>
            </div>
        )
    }
}

const mapStateToProps = (state) => {
    return {
        singleArticle: state.searchReducer.singleArticle,
        query: state.searchReducer.recentQuery,
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        searchArticle: (id) => dispatch(searchArticle(id)),
        clearArticleData : ()=>dispatch(clearArticleData())
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(SingleArticle)
