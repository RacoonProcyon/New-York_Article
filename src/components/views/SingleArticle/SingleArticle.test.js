import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import SingleArticle from './SingleArticle';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk'

const initialState = {
    searchData: [],
    pageNumber: 0,
    recentQuery: "",
    singleArticle: []
}
let middlewares = [thunk]
const mockStore = configureStore(middlewares)
let store = mockStore({ searchReducer: initialState })

it('renders correctly when there are no items', () => {
    const tree = renderer.create(<Provider store={store}>
        <SingleArticle match={{ params: { id: "123" } }} />
    </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});