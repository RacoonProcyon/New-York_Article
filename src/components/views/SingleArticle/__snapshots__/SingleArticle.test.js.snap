// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there are no items 1`] = `
<div
  className="d-flex flex-column align-items-center"
>
  <header
    className="container-fluid px-2 py-3 d-flex flex-row justify-content-center align-items-center"
    id="header"
  >
    <p
      className="text-center text-white my-auto"
    >
      "The New York Times" article search application
    </p>
  </header>
  <div
    className="mt-5 col-12 col-md-8"
  >
    <h3
      className="mt-1"
      style={
        Object {
          "fontWeight": "700",
        }
      }
    >
      HeadLine
    </h3>
    <span
      className="mt-2"
    >
      DATE
    </span>
    <p
      className="mt-3"
    >
      Details
    </p>
    <a
      href="#"
      target="_blank"
    >
      Read the full article
    </a>
  </div>
</div>
`;
